src/v2/Apps/Home/Components/HomeArtworkModuleContext.tsx

```diff
@@ -0,0 +1,249 @@
+import { Box, BaseTabs, BaseTab, Spacer, Text, Flex } from "@artsy/palette"
+import React from "react"
+import { createFragmentContainer, graphql } from "react-relay"
+import { RouterLink } from "v2/System/Router/RouterLink"
+import { HomeArtworkModuleContext_context } from "v2/__generated__/HomeArtworkModuleContext_context.graphql"
+
+interface HomeArtworkModuleContextProps {
+  title?: string | null
+  context: HomeArtworkModuleContext_context | null
+}
+
+const HomeArtworkModuleContext: React.FC<HomeArtworkModuleContextProps> = ({
+  title,
+  context,
+}) => {
+  if (!context && !title) return null
+
+  if (!context) {
+    return <Text variant="lg">{title}</Text>
+  }
+
+  switch (context.__typename) {
+    case "Fair": {
+      return (
+        <Flex justifyContent="space-between">
+          <Box>
+            <Text
+              variant="lg"
+              as={RouterLink}
+              // @ts-ignore
```

Comment:
These keep adding up lately in relation to the `as={Thing}` -- as this doesn't seem like an easy fix mind filing a JIRA ticket with what you know?

Reply:
I like this approach tho! Better than `<Link><Text>...` nesting that we've been doing elsewhere.

Reply:
The typing for `as` is sort of notoriously fickle. It *should* correctly infer types in the latest `@types/styled-components` though. Will file.